File: Solutions/8901/c++/program.cpp
// Copy from Quera

#include <iostream>
using namespace std;

int main()
{
    int n;
    char x, a, b, ans;
    cin >> n >> x;
    ans = x;
    for (int i = 0; i < n; i++)
    {
        cin >> a >> b;
        if ((ans == a) || (ans == b))
        {
            if (ans != a)
                ans = a;
            else
                ans = b;
        }
    }
    cout << ans;
    return 0;
}

File: Solutions/8901/java/program.java
// Copy from Quera

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        char x = sc.next().charAt(0);

        char ans = x;

        for (int i = 0; i < n; i ++){
            char a = sc.next().charAt(0);
            char b = sc.next().charAt(0);

            if( a == ans )
                ans = b;
            else if ( b == ans )
                ans = a;
        }
        System.out.println(ans);
    }
}